hs_core/views/utils.py

```diff
@@ -404,3 +404,251 @@ def show_relations_section(res_obj):
+                                            fed_resource_file_name_or_path=filename,
+                                            fed_resource_file_size=size)
+        elif not ResourceFile.objects.filter(object_id=resource.id,
+                                             resource_file=filename).exists():
+                ResourceFile.objects.create(content_object=resource,
+                                            resource_file=filename)
+
+
+def link_irods_folder_to_django(resource, istorage, foldername, exclude=()):
+    """
+    Recursively Link irods folder and all files and sub-folders inside the folder to Django
+    Database after iRODS file and folder operations to get Django and iRODS in sync
+    :param resource: the BaseResource object representing a HydroShare resource
+    :param istorage: IrodsStorage object
+    :param foldername: the folder name
+    :param exclude: a tuple that includes file names to be excluded from linking under the folder;
+     default is empty meaning nothing is excluded.
+    :return:
+    """
+    if resource and istorage and foldername:
+        store = istorage.listdir(foldername)
+        # add the unzipped files into Django resource model
+        for file in store[1]:
+            if file not in exclude:
+                file_path = os.path.join(foldername, file)
+                size = istorage.size(file_path)
+                link_irods_file_to_django(resource, file_path, size)
+        for folder in store[0]:
+            link_irods_folder_to_django(resource,
+                                        istorage, os.path.join(foldername, folder), exclude)
+
+
+def rename_irods_file_in_django(resource, src_name, tgt_name):
+    """
+    Rename file in Django DB after the file is renamed in Django side
+    :param resource: the BaseResource object representing a HydroShare resource
+    :param src_name: the file name to be renamed
+    :param tgt_name: the file name to be renamed to
+    :return:
+    """
+    if resource:
+        if resource.resource_federation_path:
+            res_file_obj = ResourceFile.objects.filter(object_id=resource.id,
+                                                       fed_resource_file_name_or_path=src_name)
+            if res_file_obj.exists():
+                res_file_obj[0].fed_resource_file_name_or_path = tgt_name
+                res_file_obj[0].save()
+        else:
+            res_file_obj = ResourceFile.objects.filter(object_id=resource.id,
+                                                       resource_file=src_name)
+            if res_file_obj.exists():
+                # since resource_file is a FileField which cannot be directly renamed,
+                # this old ResourceFile object has to be deleted followed by creation of
+                # a new ResourceFile with new file associated that replace the old one
+                res_file_obj[0].delete()
+                ResourceFile.objects.create(content_object=resource, resource_file=tgt_name)
+
```

Comment:
@pkdash Yes, adding new files are not changed - still using the old existing view functions. The only change is the GUI part with no backend being changed.